train.py

```diff
@@ -43,8 +43,8 @@
 
 def train(hyp, opt, device, tb_writer=None, wandb=None):
     logger.info(f'Hyperparameters {hyp}')
-    save_dir, epochs, batch_size, total_batch_size, weights, rank = \
-        Path(opt.save_dir), opt.epochs, opt.batch_size, opt.total_batch_size, opt.weights, opt.global_rank
+    save_dir, epochs, batch_size, total_batch_size, weights, rank, model_output_dir = \
+        Path(opt.save_dir), opt.epochs, opt.batch_size, opt.total_batch_size, opt.weights, opt.global_rank, opt.model_output_dir
```

Comment:
For consistency you should convert this to a `Path` object here as well.

Reply:
If I convert it here it makes it difficult to test if the argument was given as None will cause an exception and the empty string is a valid path. So either I'd have to add another argument or a awkward try catch, or the best model needs to always be outputted to 2 places, which would be better?

Reply:
I see. If you set type=Path in the call to `parser.add_argument` it will still allow it to be None but will convert it if specified. That's probably the way I would do it. I would do the same for the `--project` argument. This is a minor issue though and not worth a lot of fuss. So I approve either way.